rules/precompiled_apple_resource_bundle.bzl

```diff
@@ -0,0 +1,235 @@
+
+    # Create a file for bundletool to know what files to copy
+    # https://github.com/bazelbuild/rules_apple/blob/d29df97b9652e0442ebf21f1bc0e04921b584f76/tools/bundletool/bundletool_experimental.py#L29-L46
+    bundletool_instructions = struct(
+        bundle_merge_files = control_files,
+        bundle_merge_zips = [],
+        output = output_bundle_dir.path,
+        code_signing_commands = "",
+        post_processor = "",
+    )
+    bundletool_instructions_file = intermediates.file(
+        ctx.actions,
+        ctx.label.name,
+        "bundletool_actions.json",
+    )
+    ctx.actions.write(
+        output = bundletool_instructions_file,
+        content = bundletool_instructions.to_json(),
+    )
+    ctx.actions.run(
+        executable = ctx.executable._bundletool_experimental,
+        mnemonic = "BundleResources",
+        progress_message = "Bundling " + bundle_name,
+        inputs = input_files + [bundletool_instructions_file],
+        outputs = [output_bundle_dir],
+        arguments = [bundletool_instructions_file.path],
+    )
+    return [
+        AppleResourceInfo(
+            unowned_resources = depset(),
+            owners = depset([
+                (output_bundle_dir.short_path, ctx.label),
+                (output_plist.short_path, ctx.label),
+            ]),
+            # This is a list of the resources to propagate without changing further
+            # In this case the tuple parameters are:
+            # 1. The final directory the resources should end up in, ex Foo.bundle
+            #    would result in Bar.app/Foo.bundle
+            # 2. The Swift module associated with the resources, this isn't
+            #    required for us since we don't use customModuleProvider in IB
+            # 3. The resources to propagate, in our case this is just the final
+            #    Foo.bundle directory that contains our real resources
+            unprocessed = [
+                (output_bundle_dir.basename, None, depset([output_bundle_dir, output_plist])),
+            ],
+        ),
+        AppleResourceBundleInfo(),
+        apple_common.new_objc_provider(),
+    ]
+
+precompiled_apple_resource_bundle = rule(
+    implementation = _precompiled_apple_resource_bundle_impl,
+    fragments = ["apple"],
+    cfg = transition_support.apple_rule_transition,
+    attrs = dict(
+        # This includes all the undocumented tool requirements for this rule
+        rule_factory.common_tool_attributes,
+        infoplists = attr.label_list(
+            allow_files = [".plist"],
+            default = [
+                Label("@build_bazel_rules_ios//rules/library:resource_bundle.plist"),
+            ],
+        ),
+        bundle_name = attr.string(
+            mandatory = False,
+            doc = "The name of the resource bundle. Defaults to the target name.",
+        ),
+        bundle_id = attr.string(
+            mandatory = False,
+            doc = "The bundle identifier of the resource bundle.",
```

Comment:
Or if we are providing a default (which I suggest we don't) it should be documented here.